docs/development/external-name-handling.md

```diff
@@ -0,0 +1,128 @@
+# External Name Handling
+
+## Context
+`crossplane.io/external-name` is an annotation for Managed Resources. Across providers we lack a unified definition and implementation guideline. This ADR proposes a consistent definition, expected behavior, and implementation approach.
+
+## Background
+
+### Definitions
+
+The external-name (string) identifies the external resource in the external system.
+Based on this definition, common assumptions are:
+1. Human-facing name of the resource shown in the external provider UI. [1]
+2. (ideally Unique) identifier generated by the external system (GUID). [2]
+
+### Default Behavior in Crossplane
+- By default, the external name is initialized (by crossplane-runtime) to the Kubernetes `metadata.name` of the Managed Resource (MR) if not explicitly set.
+- This default conflicts with cases where the external system uses a different unique identifier.
+
+### Importing Existing External Resources (Official Flow) [3]
+1. Create an MR manifest with `spec.managementPolicies: Observe`.
+2. Add the external-name annotation with the external resource identifier. If not globally unique, also supply distinguishing `spec.forProvider` fields.
+3. Apply the MR. Controller populates `status.atProvider`.
+4. Copy the fields from `status.atProvider` to `spec.forProvider` and change `managementPolicies` to `*` to gain full control.
+
+## Our Approach
+
+### Definition
+The external-name (string) identifies the external resource in the external system. 
+A provider can adopt two qualities:
+
+(1) The existance of the external-name express that there should be an external resource it identifies. So if the user creates a MR with the external-name set, the user explicitly states that this resource already exists in the external system and should be adopted/matched. This prevents unintended adoption/matching of existing resources. (2) The identifier than contains the information the provider needs to identify the external resource. This can be an unique identifier or a compound key.
```

Comment:
Dev Sync: 
- less strict about knowledge of users
- intent via mangement policies doesnt cover the case of accidently mathcing resource